Util/src/main/java/io/deephaven/util/thread/ThreadInitializationFactory.java

```diff
@@ -0,0 +1,44 @@
+package io.deephaven.util.thread;
+
+import io.deephaven.configuration.Configuration;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Extension point to allow threads that will run user code from within the platform to be controlled by configuration.
+ */
+public interface ThreadInitializationFactory {
+    /* private */ String[] CONFIGURED_INITIALIZATION_TYPES =
+            Configuration.getInstance().getStringArrayFromProperty("thread.initialization");
```

Comment:
Does this work if the property doesn't exist or is empty?

Reply:
It fails if it does not exist, but succeeds if empty (with an empty array).

Due to how defaults work (or rather, don't work), I was under the impression we wanted to generally avoid using them, so dh-defaults.prop now contains this property.